src/shared/lending/oracles/FortressWstETHwETHOracle.sol

```diff
@@ -23,40 +23,85 @@ import {IBalancerV2StablePool} from "../interfaces/IBalancerV2StablePool.sol";
+contract FortressWstETHwETHOracle {
 
     using FixedPointMathLib for uint256;
 
+    uint256 public lastSharePrice;
+    uint256 public lowerBoundPercentage;
+    uint256 public upperBoundPercentage;
+    uint256 public vaultMaxSpread;
+
+    address public owner;
+    address public vault;
+
     uint256 public ethUSDFeed_decimals = 1e8;
     uint256 public wstEthETHFeed_decimals = 1e18;
 
+    bool public isCheckPriceDeviation;
+
+    uint256 constant internal _BASE = 1e18;
+
     IChainlinkAggregator public ethUSDFeed = IChainlinkAggregator(address(0x639Fe6ab55C921f74e7fac1ee960C0B6293ba612));
     IChainlinkAggregator public wstEthETHFeed = IChainlinkAggregator(address(0xb523AE262D20A936BC152e6023996e46FDC2A95D));
     IBalancerV2StablePool public BPT = IBalancerV2StablePool(address(0x36bf227d6BaC96e2aB1EbB5492ECec69C691943f));
 
     /********************************** Constructor **********************************/
 
-    constructor(address _owner, address _vault) BaseOracle(_owner, _vault) {}
+    constructor(address _owner, address _vault) {
+        lowerBoundPercentage = 20;
+        upperBoundPercentage = 20;
+        
+        owner = _owner;
+        vault = _vault;
+
+        uint256 _vaultSpread = ERC4626(_vault).convertToAssets(1e18);
+        vaultMaxSpread = _vaultSpread * 110 / 100; // limit to 10% of the vault spread
+
+        lastSharePrice = uint256(_getPrice());
+
+        isCheckPriceDeviation = true;
+    }
 
     /********************************** Modifiers **********************************/
 
     modifier reentrancyCheck() {
         IBalancerVault(BPT.getVault()).manageUserBalance(new IBalancerVault.UserBalanceOp[](0));
         _;
     }
+    
+    modifier onlyOwner() {
+        if (msg.sender != owner) revert notOwner();
+        _;
+    }
 
     /********************************** External Functions **********************************/
 
-    function description() external pure override returns (string memory) {
+    function decimals() external pure virtual returns (uint8) {
+        return 18;
+    }
+
+    function version() external pure virtual returns (uint256) {
+        return 1;
+    }
+
+    function description() external pure returns (string memory) {
         return "fcwstETHwETH USD Oracle";
     }
 
+    function getRoundData(uint80) external pure returns (uint80, int256, uint256, uint256, uint80) {
+        revert("Not implemented");
+    }
+
+    function latestRoundData() external returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound) {
+        return (0, _getPrice(), 0, 0, 0);
+    }
+
     /********************************** Internal Functions **********************************/
 
-    function _getPrice() internal view override  returns (int256) {
+    function _getPrice() internal reentrancyCheck returns (int256) {
         uint256 rate = BPT.getRate();
         if (rate < 1*1e18 || rate >= 1.1*1e18)  revert virtualPriceOutOfBounds();
```

Comment:
perhaps the bounds should not be hardcoded, and give owner ability to change them?